analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -168,133 +144,275 @@ public static void LoopWithBreak(System.Collections.Generic.IEnumerable<string>
+    bool HasValue1(bool? b) => b.HasValue ? b.Value : false;                // Compliant
+    bool HasValue2(bool? b) => b.HasValue ? true : b.Value;                 // Noncompliant
+
+    bool TruthAndAssignment(bool? b) =>
+        true || ((b = null) == null) ? b.Value : false;                     // FN
+}
+
+class Linq
+{
+    private IEnumerable<TestClass> Numbers = new[]
     {
-        //HasValue and NoValue constraints are set here
-    }
-    if (foo == value)   // Relationship is added here
+        new TestClass() { Number = 42 },
+        new TestClass(),
+        new TestClass() { Number = 1 },
+        new TestClass() { Number = null }
+    };
+
+    IEnumerable<int> EnumerableOfEmptyNullableValues1 =>
+        Numbers.Where(x => !x.Number.HasValue).Select(x => x.Number.Value); // FN
+
+    IEnumerable<int> EnumerableOfEmptyNullableValues2 =>
+        Numbers.Select(x => null as int?).Select(x => x.Value);             // FN
+
+    class TestClass
     {
-        if (foo.HasValue)
-        {
-            Console.WriteLine(foo.Value.ToString());
-        }
-        if (!foo.HasValue)
-        {
-            Console.WriteLine(foo.Value.ToString());    // FIXME Non-compliant
-        }
-        if (foo == null)
-        {
-            Console.WriteLine(foo.Value.ToString());    // FN
-        }
-        if (foo != null)
-        {
-            Console.WriteLine(foo.Value.ToString());
-        }
+        public int? Number { get; set; }
     }
 }
 
-public void NestedIsolated1(DateTime? value)
+interface IWithDefaultImplementation
 {
-    if (foo == value)   // Relationship is added here
+    int DoSomething()
     {
-        if (value.HasValue)
-        {
-            if (!foo.HasValue)
-            {
-                Console.WriteLine(foo.Value.ToString());    // Compliant
-            }
-        }
+        int? i = null;
+        return i.Value; // Noncompliant
     }
 }
 
-public void NestedIsolated2(DateTime? value)
+// https://github.com/SonarSource/sonar-dotnet/issues/4573
+class Repro_4573
 {
-    if (foo == value)   // Relationship is added here
+    private DateTime? foo;
+
+    public virtual DateTime? Foo
     {
-        if (!value.HasValue)
+        get => foo;
+        set
         {
-            if (foo == null)
+            if (value.HasValue)
+            {
+                //HasValue and NoValue constraints are set here
+            }
+            if (foo != value || foo.HasValue)
             {
-                Console.WriteLine(foo.Value.ToString());    // FIXME Non-compliant
+                foo = value;
             }
         }
     }
-}
 
-public void NestedCombined(DateTime? value)
-{
-    if (foo == value)   // Relationship is added here
+    public void Sequence(DateTime? value)
     {
         if (value.HasValue)
+        {
+            // HasValue and NoValue constraints are set here
+        }
+        if (foo == value)   // Relationship is added here
```

Comment:
As relationships are not supported by the engine, we need to update all these comments (also below)
```suggestion
        if (foo == value)   // Relationship should be created here
```